### PR TITLE
Policies

### PR DESCRIPTION
**Pull request by josedonizetti**
### 1. Explain what the PR does

This PR adds policies support (https://github.com/aquasecurity/tracee/issues/2665). 

We have an intermediate structure (FilterMap) which is generated from policy files or flags, this structure than is used to create the internal policies tracee uses to filter events. There are a few examples under a directory which can be used for testing and also to get a grasp of how policies work, also take a look at https://github.com/aquasecurity/tracee/discussions/2640

For now, we are not supporting any actions other than `log`, `webhook` and `fluentd` are coming in a separate PR, and also not supporting actions override under the events structure.

### 2. Explain how to test it

```
tracee -p examples/new_containers.yaml # only one policy
tracee -p examples # directory with several policies
tracee -p examples/dig.yaml -p examples/new_containers.yaml # specific policies
```

### CODE REVIEW

File: examples/uid_higher_or_equal_than_zero.yaml
Comment (geyslan) [inline examples/uid_higher_or_equal_than_zero.yaml:2]:
```suggestion
name: uid_equal_to_or_higher_than_zero
description: This policy traces read and write syscall for uids equal to or higher than 0
```
rename the file name as well.

File: pkg/cmd/urfave/urfave.go
Comment (geyslan) [inline pkg/cmd/urfave/urfave.go:244]: path is empty, so the formatting is not necessary. Perhaps %d (index)?

File: pkg/cmd/urfave/urfave.go
Comment (geyslan) [inline pkg/cmd/urfave/urfave.go:119]:
```suggestion
		filterMap, err = flags.PrepareFilterMapFromPolicies(policies)
```

WDYT?

File: pkg/cmd/urfave/urfave.go
Comment (geyslan) [inline pkg/cmd/urfave/urfave.go:124]: ditto

```suggestion
		filterMap, err = flags.PrepareFilterMapFromFlags(c.StringSlice("filter"))
```

**Comment by geyslan** [comment]
Ran a few tests and all seemed good.

File: examples/uid_higher_or_equal_than_zero.yaml
Comment (josedonizetti) [reply examples/uid_higher_or_equal_than_zero.yaml:2]: why? if the signal is `>=` higher or equal

**Review by geyslan — APPROVED** [review]
LGTM

**Comment by yanivagman** [comment]
I want to review this one before merge

**Comment by geyslan** [comment]
> I want to review this one before merge

On top of this we have this other PR:

- https://github.com/josedonizetti/tracee/pull/2

File: examples/context_processname.yaml
Comment (yanivagman) [inline examples/context_processname.yaml:9]: let's keep the context filters uniform with the scope context

File: examples/pid_lower_than_1000.yaml
Comment (yanivagman) [inline examples/pid_lower_than_1000.yaml:2]: actually pids can wrap, so better use equality (`pid=1000`) instead of inequality in this example

File: examples/pid_lower_than_1000.yaml
Comment (yanivagman) [inline examples/pid_lower_than_1000.yaml:12]: just to justify the use of event context filter and not a scope, better have different pid filters for the read and write events

File: examples/uid_higher_than_or_equal_than_zero.yaml
Comment (yanivagman) [inline examples/uid_higher_than_or_equal_than_zero.yaml:2]: nit: greater than, equal to

File: examples/uid_higher_than_or_equal_than_zero.yaml
Comment (yanivagman) [inline examples/uid_higher_than_or_equal_than_zero.yaml:4]: the file name and description say 0, but the filter has 1000

File: pkg/cmd/urfave/urfave.go
Comment (yanivagman) [inline pkg/cmd/urfave/urfave.go:278]: `.yml` should also be an option

File: pkg/cmd/flags/policy.go
Comment (yanivagman) [inline pkg/cmd/flags/policy.go:105]: we can add a note that currently only one rule per event is supported (we may add support for more than one rule per event in the future)

File: pkg/cmd/flags/policy.go
Comment (yanivagman) [inline pkg/cmd/flags/policy.go:40]: I think we are missing validation that the same policy name is not used more than once

**Review by yanivagman — CHANGES_REQUESTED** [review]
Can't wait for this to be merged :-)
A few comments inline

File: pkg/cmd/flags/policy.go
Comment (yanivagman) [reply pkg/cmd/flags/policy.go:114]: Actually I meant that the error message should indicate that, something like:
"policy x uses event y in more than one rule which is not currently supported"

**Review by yanivagman — APPROVED** [review]
Nit comment, otherwise LGTM

File: pkg/cmd/flags/policy.go
Comment (josedonizetti) [reply pkg/cmd/flags/policy.go:114]: AHHHHH! sorry, fixing it

**Comment by geyslan** [comment]
Since we now have this merged, I have josedonizetti#2 and #2919 to rebase, but depending on the last step of the #2760 types changes and that github is back up and running.

**Pull request merged**
